source/common/router/config_impl.cc

```diff
@@ -154,6 +161,17 @@ bool RouteEntryImplBase::matchRoute(const Http::HeaderMap& headers, uint64_t ran
 const std::string& RouteEntryImplBase::clusterName() const { return cluster_name_; }
 
 void RouteEntryImplBase::finalizeRequestHeaders(Http::HeaderMap& headers) const {
+  // Add user-specified request headers from the vhost and then from the route.
```

Comment:
I'm not super thrilled that we do one thing in the connection manager and two things here. I'm wondering if we should move at least adding request headers for the global route level here. (We can leave munging response headers in the connection manager). Thoughts?